uaclient/api/u/pro/services/list/v1.py

```diff
@@ -0,0 +1,212 @@
+        desc: str,
+        available: bool,
+        entitled: Optional[bool] = None
+    ):
+        self.name = name
+        self.description = desc
+        self.available = available
+        self.entitled = entitled
+
+
+class ServiceListResult(DataObject, AdditionalInfo):
+    fields = [
+        Field("services", data_list(ServiceInfo), doc="List of services"),
+    ]
+
+    def __init__(self, *, services: List[ServiceInfo]):
+        self.services = services
+
+
+def list() -> ServiceListResult:
+    return _list(UAConfig())
+
+
+def _list(cfg: UAConfig) -> ServiceListResult:
+    if not _is_attached(cfg).is_attached:
+        return unattached_services(cfg)
+    return attached_services(cfg)
+
+
+def attached_services(cfg: UAConfig) -> ServiceListResult:
+    services_list = []
+    machine_token_file = machine_token.get_machine_token_file(cfg)
+    resources = machine_token_file.machine_token.get("availableResources")
+    if not resources:
+        resources = get_available_resources(cfg)
+    for resource in resources:
+        try:
+            ent = entitlement_factory(cfg=cfg, name=resource.get("name", ""))
+        except exceptions.EntitlementNotFoundError:
+            LOG.debug(
+                "Ignoring availability of unknown service %s from contract "
+                "server",
+                resource.get("name", "without a 'name' key"),
+            )
+            continue
+
+        services_list.append(_get_service_info(cfg, ent))
+    services_list.sort(key=lambda x: x.name)
+    return ServiceListResult(services=services_list)
+
+
+def unattached_services(cfg: UAConfig) -> ServiceListResult:
+    services_list = []
+    resources = get_available_resources(cfg)
+    for resource in resources:
+        available = True if resource.get("available") else False
+        try:
+            ent = entitlement_factory(cfg=cfg, name=resource.get("name", ""))
+        except exceptions.EntitlementNotFoundError:
+            LOG.debug(
+                "Ignoring availability of unknown service %s from contract "
+                "server",
+                resource.get("name", "without a 'name' key"),
+            )
+            continue
+        service = ServiceInfo(
+            name=resource.get("presentedAs", resource["name"]),
+            desc=ent.description,
+            available=available,
+            entitled=None,
+        )
+        services_list.append(service)
+    services_list.sort(key=lambda x: x.name)
+    return ServiceListResult(services=services_list)
+
+
+def _get_service_info(
+    cfg: UAConfig,
+    ent: UAEntitlement,
+) -> ServiceInfo:
+    contract_status = ent.contract_status()
+    entitled = None
+    if contract_status == ContractStatus.UNENTITLED:
+        entitled = False
+    elif contract_status == ContractStatus.ENTITLED:
+        entitled = True
+    available = (
+        True
+        if ent.applicability_status()[0] == ApplicabilityStatus.APPLICABLE
+        else False
+    )
```

Comment:
No need for an if-else
```suggestion
    available = ent.applicability_status()[0] == ApplicabilityStatus.APPLICABLE
```